src/binary-reader.cc

```diff
@@ -1454,6 +1449,9 @@ Result BinaryReader::ReadRelocSection(Offset section_size) {
 
 Result BinaryReader::ReadLinkingSection(Offset section_size) {
   CALLBACK(BeginLinkingSection, section_size);
+  uint32_t version;
+  CHECK_RESULT(ReadU32Leb128(&version, "version"));
```

Comment:
FWIW I think this breaks using tools like `wasm2wat` with code generated by rustc today. Rustc is still on LLVM 6 (and the LLD at the time) which I don't think encodes the version here, so this is causing `wasm2wat` to fail on any rustc-generated binary (which currently retain a small `linking` section).

Would it be possible to ignore errors here or otherwise not fail decoding if this fails? According to the [binary encoding spec](https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md#name-section):

> If a WebAssembly implementation interprets the payload of any custom section during module validation or compilation, errors in that payload must not invalidate the module.

Reply:
Ah, thanks for mentioning this. We've discussed in the past that it would be best to have an option to require or ignore errors in custom sections, and this is a good example of why that's needed.

I'll see about adding that today.

Reply:
Awesome, thanks!

I'm also hoping we can figure out a strategy for ourselves to upgrade LLVM in the meantime :)

Reply:
This landed here: https://github.com/WebAssembly/wabt/commit/9890d45ff4e04a89ba53137d5f569c5038df8d27

You can use the `--ignore-custom-section-errors` flag in `wasm2wat` or `wasm-validate` for this.

Reply:
Awesome, thanks!